llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5012,6 +5012,10 @@ bool SelectionDAG::isGuaranteedNotToBeUndefOrPoison(SDValue Op,
   case ISD::TargetFrameIndex:
     return true;
 
+  case ISD::CopyFromReg:
+    // Treat CopyFromReg as freezing the value.
```

Comment:
Just add to the case above handling FrameIndex and co? Don't need the special comment?